packages/gator-permissions-snap/src/core/accountController.ts

```diff
@@ -168,4 +182,75 @@ export class AccountController {
 
     return { domain, types, primaryType, message, metadata };
   }
+
+  /**
+   * Checks if the account is already upgraded to a smart account.
+   * @param params - The account and chain ID to check.
+   * @returns Promise resolving to the upgrade status.
+   */
+  public async getAccountUpgradeStatus(
+    params: AccountUpgradeParams,
+  ): Promise<AccountUpgradeStatus> {
+    logger.debug('AccountController:getAccountUpgradeStatus()', params);
+
+    try {
+      const result = (await this.#ethereumProvider.request({
+        method: 'wallet_getAccountUpgradeStatus',
+        params: [params],
+      })) as { isUpgraded: boolean; upgradedAddress: Hex | null };
+
+      logger.debug('Account upgrade status result', result);
+
+      const {
+        contracts: { eip7702StatelessDeleGatorImpl },
+      } = getChainMetadata({ chainId: hexToNumber(params.chainId) });
+
+      return {
+        isUpgraded:
+          result.isUpgraded &&
+          result.upgradedAddress?.toLowerCase() ===
+            eip7702StatelessDeleGatorImpl.toLowerCase(),
+      };
+    } catch (error) {
+      logger.error('Failed to check account upgrade status', error);
+      throw new InternalError('Failed to check account upgrade status');
+    }
+  }
+
+  /**
+   * Upgrades the account to a smart account.
+   * @param params - The account and chain ID to upgrade.
+   * @returns Promise resolving to the upgrade result with transaction hash.
+   */
+  public async upgradeAccount(
+    params: AccountUpgradeParams,
+  ): Promise<AccountUpgradeResult> {
+    logger.debug('AccountController:upgradeAccount()', params);
+
+    try {
+      const result = await this.#ethereumProvider.request({
+        method: 'wallet_upgradeAccount',
+        params: [params],
+      });
+
+      logger.debug('Account upgrade result', result);
+
+      // The result should contain a transaction hash
+      if (
+        typeof result === 'object' &&
+        result !== null &&
+        'transactionHash' in result
+      ) {
+        return {
+          transactionHash: (result as { transactionHash: string })
+            .transactionHash,
+        };
+      }
```

Comment:
If you use the generic param on the `request` call above you can simplify this to:
```suggestion
      if (result?.transactionHash) {
        return {
          transactionHash: result.transactionHash,
        };
      }
```